pkg/auth/auth.go

```diff
@@ -38,19 +36,14 @@ func GetServicePrincipalTokenFromMSI(resource string) (_ *adal.Token, err error)
 		}
 	}()
 
-	msiEndpoint, err := adal.GetMSIVMEndpoint()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get the MSI endpoint, error: %+v", err)
-	}
-	// Set up the configuration of the service principal
-	spt, err := adal.NewServicePrincipalTokenFromMSI(msiEndpoint, resource)
+	spt, err := adal.NewServicePrincipalTokenFromManagedIdentity(resource, nil)
 	if err != nil {
-		return nil, fmt.Errorf("failed to acquire a token for MSI, error: %+v", err)
+		return nil, err
```

Comment:
Is there a reason why we remove the description of the error? I found them pretty helpful when debugging and determining where the error is coming from.

Reply:
From initial glance, it looks like we're wrapping the error at every stage so decided to remove it from here. I can add it back for now.

Reply:
Didn't realize that. Feel free to remove those that are duplicated. Thanks!